### PR TITLE
Implement SED secrets

### PR DESCRIPTION
**Pull request by sadlerap**
SED information could contain sensitive data, such as a password or authentication token.  Such data should not be stored within a registered service's resource itself.  When making registered services with a service class, we should give users the option to indicate that their data is secret.

This change introduces a `Secret` flag within a SED mapping, which indicates whether that particular mapping item should be stored within the registered service (false) or within an associated secret (true). By default, this flag is true, since security features such as this should be opt-out, not opt-in.

### CODE REVIEW

File: controllers/agents/svc/serviceclass_controller.go [hunk 1]
```diff
@@ -106,15 +112,24 @@ func (r *ServiceClassReconciler) Reconcile(ctx context.Context, req ctrl.Request
...
 				reconcileLog.Error(err, "Failed to create registered service",
 					"service", rs.Name,
 					"namespace", rs.Namespace)
 			} else {
 				reconcileLog.Info("Wrote registered service", "service", rs.Name, "namespace", rs.Namespace, "operation", op)
 			}
+
+			if secret != nil {
+				_, err1 := controllerutil.CreateOrUpdate(ctx, remote_client, secret, func() error { return controllerutil.SetOwnerReference(clone, secret, remote_client.Scheme()) })
+				err = errors.Join(err, err1)
```
Comment (Kartikey-star) [inline controllers/agents/svc/serviceclass_controller.go:131]: We seem to be preparing the error List on L299, so we can avoid this operation here right?

File: controllers/agents/svc/serviceclass_controller.go [hunk 2]
```diff
@@ -130,15 +145,20 @@ func (r *ServiceClassReconciler) Reconcile(ctx context.Context, req ctrl.Request
...
 				if apierrors.IsNotFound(err) {
 					// we tried to delete an object that doesn't exist, so
 					return nil
 				}
 				reconcileLog.Error(err, "Failed to delete registered service", "namespace", rs.Namespace)
+				return err
 			}
-			return err
+
+			if secret != nil {
```
Comment (Kartikey-star) [inline controllers/agents/svc/serviceclass_controller.go:158]: We are setting the Registered Service as owner of the secret in L130, so can deletion of secret be avoided here?

File: controllers/agents/svc/serviceclass_controller.go (hunk 1 above)
Comment (sadlerap) [reply controllers/agents/svc/serviceclass_controller.go]: I've changed the implementation to aggregate errors there.

File: controllers/agents/svc/serviceclass_controller.go (hunk 2 above)
Comment (sadlerap) [reply controllers/agents/svc/serviceclass_controller.go]: Yep, good catch.  Removed.

**Review by Kartikey-star — APPROVED** [review]
Looks good to me

**Pull request merged**